code/site/components/com_pages/dispatcher/router/abstract.php

```diff
@@ -87,23 +87,18 @@ protected function _initialize(KObjectConfig $config)
      */
     public function resolve($route, array $parameters = array())
     {
-        $result = false;
-
         $route = $this->getRoute($route, $parameters);
 
         if(!$route->isResolved())
         {
             $this->__resolvers->setIteratorMode(SplDoublyLinkedList::IT_MODE_LIFO | SplDoublyLinkedList::IT_MODE_KEEP);
 
-            foreach($this->__resolvers as $resolver)
-            {
-                if(false === $result = $resolver->resolve($route, $parameters)) {
-                    break;
-                }
+            foreach($this->__resolvers as $resolver) {
+                $resolver->resolve($route, $parameters);
```

Comment:
@johanjanssens shouldn't we add a check for `$route->isResolved()` here?

Reply:
No because the route can be resolved be the first resolver and is then further manipulated by subsequent resolvers in the queue. This is how i do pagination. 

It's up to each resolver to decide if how to do this. if the route has been resolved they can skip.